scripts/extensions/sams/src/components/assets/assetFilterPanel.tsx

```diff
@@ -79,6 +87,54 @@ export class AssetFilterPanel extends React.PureComponent<IProps, IState> {
         };
     }
 
+    setLocalAssetTagSearchParams(tag: IAssetTag, method?: string) {
```

Comment:
This code is very similar to the code we already have in the `AssetEditor` component.

To help maintain this code (and reduce duplicated code), it would be ideal if we had a React component for these Asset Tag Inputs.

Using a similar interface to the UI-Framework input fields, the props could include:
```ts
interface IProps {
    label?: string;  // defaults to `gettext('Tags')
    value?: Array<IAssetTag>; // initial value
    onChange(value: Array<IAssetTag>): void; // provides the final Array value to be stored with the Asset
    disabled?: boolean;
    existingTagsOnly?: boolean;
}
```

This new component would handle internally the following:
* Searching for tags (i.e. via `samsApi.assets.searchTags`)
* Using `addTag` and `removeTag` to modify this components internal state (and calling `onChange` with the new Array)
* Rendering the Input field
* Rendering the list of Tags

There were some improvements done to this input type in `AssetEditor`, so it might be good to look at both of these implementations to come up with the final implementation

Reply:
In here for `setLocalAssetTagSearchParams`, the issue is that states are diff for both filter and editor, due to which I did not use same function for both of them.
Rest things are mostly same.

Reply:
The `onChange` prop of this new component would always provide an `Array<IAssetTag>`. It is then up to the consuming component (`AssetEditor` or `AssetFilterPanel`) to store these tags whichever way is required. This way the functionality is generic inside this new component, and we use `onChange` to control where the tags are stored

Reply:
This improvement will be added at a later date